### ISSUE TITLE
Overlapping markers cause the wrong item to be selected when using: onItemLongPress()

### ISSUE THREAD

**Issue by openBDM** [osmdroid/osmdroid#1122]
## Issue Type

[x ] Question
[x ] Bug
[ ] Improvement
[ ] Build system related
[ ] Performance
[ ] Documentation

## Description and/or steps/code to reproduce the problem
![image](https://user-images.githubusercontent.com/41287133/43583475-e6684770-965f-11e8-8897-b355d04ad80f.png)
When such as shown above markers are set on a map and the following code is used: 

```java
ItemizedOverlayWithFocus<OverlayItem> mOverlay = new ItemizedOverlayWithFocus<OverlayItem> 
   (this,items,
                new ItemizedIconOverlay.OnItemGestureListener<OverlayItem>() {
                    @Override
                    public boolean onItemSingleTapUp(final int index, final OverlayItem item) {
                        return true;
                    }
                    @Override
                    public boolean onItemLongPress(final int index, final OverlayItem item) {
                        System.out.println("TWICE");
                    
                        return false;
                    }
                });
        mOverlay.setFocusItemsOnTap(true);

        map.getOverlays().add(mOverlay);
```
 

It will open the item in the back (the one not selected/longpressed)

How could I solve this?
## Environment
Nexus 5 android emulator

### If it's a bug, version(s) of android this affects:

All

### Version of osmdroid the issue relates to:

6.0.2

**Comment by monsieurtanuki** [comment]
I don't know if the "wrong" marker is selected (you prefer the top one, don't you?), but this is how I handle such cases:
* I use a `List<OverlayItem>`
* in `ItemizedOverlayWithFocus<>.onItemSingleTapUp`, I populate this `List` and always return false, so that all the `OverlayItem`s are checked, and I will have to check the other overlays afterwards
* I have a `MapEventsReceiver` before the `ItemizedOverlayWithFocus<>` (therefore its `onItemSingleTapUp` is called after), which checks this list (and `clear` it afterwards)
* this way I can handle the 0, 1 or n `OverlayItem`(s) clicked upon the way I want it (in my app, when there are several clicked items I display a list via an `Adapter` in order to select the actual item - I may want the hidden ones)
* you can also do it the way I presume you want it - if there are n items you get the last one (supposedly the last being displayed)

Tell me if it works for you.

**Comment by copypasteearth** [comment]
@monsieurtanuki  do you by any chance have a way to do it with OnMarkerClickListener?

**Comment by monsieurtanuki** [comment]
@copypasteearth I don't have access to my laptop now, but the logic should be the same with markers. I'll create a demo for both by the end of August.

**Comment by copypasteearth** [comment]
ahhh, thanks I got it working how you suggested above but the only thing is that MapEventsOverlay is depreciated. Im just worried about that. Is it ok to still use it and put it into production haha?

**Comment by monsieurtanuki** [comment]
The `MapEventsOverlay` class is _not_ `deprecated`.
Its `(Context, MapEventsReceiver)` constructor is indeed `deprecated`, but the other `(MapEventsReceiver)` constructor isn't.

**Comment by monsieurtanuki** [comment]
@openBDM @copypasteearth I've just merged new samples in #1150.
Does that give you inspiration to solve your issues?

**Comment by monsieurtanuki** [comment]
@openBDM @copypasteearth ping

**Comment by copypasteearth** [comment]
Sorry for no responses on my issues. I am in school and i wont get back to my project with maps until late spring.

**Comment by monsieurtanuki** [comment]
@openBDM I think the fix to your OP would be to edit `ItemizedIconOverlay.activateSelectedItems` in order to list the items _in a reverse order_, a bit like what is done in `DefaultOverlayManager.onLongPress`.
But that doesn't explain why you currently have different results between single tap and long press.
Feel free to PR.

**Comment by monsieurtanuki** [comment]
Looking back at the issue, I think the fix is only to return `true` in method ` onItemLongPress`.
Which for some reason was not the case in the samples, and in the OP code excerpt.
I "fixed" the samples of the demo app, assuming that the expected behavior for long clicks is the same as for short clicks: identify _only_ the first item being clicked on.
I've just merged #1262.
Closing the issue now; @openBDM feel free to reopen the issue if necessary.
